Makefile

```
@@ -39,8 +30,18 @@ GINKGO        := ginkgo
 RESOLVED_PWD  := $(shell realpath $(shell pwd))
 BUILD_PATH    := $(RESOLVED_PWD)/build
 BIN_PATH      := $(BUILD_PATH)/bin
+TOOLS_PATH    := $(BUILD_PATH)/tools
 COVERAGE_PATH := $(BUILD_PATH)/coverage
 
+PROTO_SOURCES = $(shell find pkg -name '*.proto' | grep -v /vendor/)
+PROTO_GOFILES = $(patsubst %.proto,%.pb.go,$(PROTO_SOURCES))
+PROTO_INCLUDE = -I $(PWD) -I$(TOOLS_PATH)/include
+PROTO_OPTIONS = --proto_path=. $(PROTO_INCLUDE) \
+    --go_opt=paths=source_relative --go_out=. \
+    --go-ttrpc_opt=paths=source_relative --go-ttrpc_out=. \
+    --go-plugin_opt=paths=source_relative,disable_pb_gen=true --go-plugin_out=.
+PROTO_COMPILE = PATH=$(TOOLS_PATH)/bin protoc $(PROTO_OPTIONS)
```

```suggestion
PPROTO_COMPILE = PATH="$(TOOLS_PATH)/bin:$$PATH" protoc $(PROTO_OPTIONS)
```
Isn't this necessary ?

---

No. At least for me it works fine without. And makes sure that no plugins are accidentally taken from the system